Lib/Internal.ps1

```diff
@@ -223,3 +227,21 @@ function CopyDirectory {
         Write-Progress -Activity $activity -Completed;
     } #end process
 } #end function CopyDirectory
+
+function TestComputerName {
+<#
+    .SYNOPSIS
+        Validates a computer name is valid.
+#>
+    [CmdletBinding()]
+    [OutputType([System.Boolean])]
+    param (
+        ## Source directory path
+        [Parameter(Mandatory, ValueFromPipeline)] [ValidateNotNullOrEmpty()]
+        [System.String] $ComputerName
+    )
+    process {
+        $invalidMatch = '[~!@#\$%\^&\*\(\)=\+_\[\]{}\\\|;:.''",<>\/\?\s]';
+        return ($ComputerName -inotmatch $invalidMatch);
```

Comment:
There's also a 15 char limit on the computer name. You'll continue to get an error in the answer file if your computer name is long than 15 characters.

![image](https://user-images.githubusercontent.com/3868453/30037650-03947f2e-918b-11e7-93af-46424375a52f.png)

It might be useful to check for that as well in the same code block.